pages/activity/[id]/agenda/[agendaId]/invitation.tsx

```diff
@@ -17,59 +18,70 @@ import styles from '../../../../../styles/invitation.module.less';
 
 export const getServerSideProps: GetServerSideProps<
   { activity: Activity; agenda: Agenda },
-  { id: string; agendaId: string }
+  { id: string; agendaId: string },
+  { currentUrl: string }
 > = async context => {
   const activityStore = new ActivityModel();
   const { id, agendaId } = context.params!;
   const activity = await activityStore.getOne(id + '');
   const agenda = await activityStore.currentAgenda!.getOne(agendaId + '');
+  const { host, referer } = context.req.headers;
+  const currentUrl = `${referer ? new URL(referer).origin : `http://${host}`}${
+    context.req.url
+  }`;
```

Comment:
先导出一个常量：
https://github.com/kaiyuanshe/kaiyuanshe.github.io/blob/1563b159ce602ec9576b186d5ab5a49ade789762/models/Base.ts#L8

再改为：
```suggestion
> = async ({ resolvedUrl, params, }) => {
  const activityStore = new ActivityModel(),
      { id, agendaId } = params!;
  const activity = await activityStore.getOne(id + '');
  const agenda = await activityStore.currentAgenda!.getOne(agendaId + '');
  const currentUrl = API_HOST + resolvedUrl;
```